text/mavlink2_1.md

```diff
@@ -0,0 +1,62 @@
+  * Start date: 2022-10-12
+  * Contributors: Andrew Tridgell
+
+# Summary
+
+Rough proposal for MAVLink 2.1
+
+# Motivation
+
+Key features wanted:
+
+ * support for more than 256 system IDs
+ * support for checking CRCs in routers that don't have all xml
+ * better target_system/target_component support
+ * maximum compatibility with existing mavlink2 systems
+
+# Detailed Design
+
+The design takes advantage of the incompat_flags and compat_flags
+fields in the MAVLink2 header.
+
+New incompat_flags flags would be:
+
+ * MAVLINK_IFLAG_SYSID16 : system ID in header is 16 bits wide
+ * MAVLINK_IFLAG_TARGETTED : header has target_system and target_component appended
+ * MAVLINK_IFLAG_SPLIT_CRC : 16 bit crc is as described below
+
+New compat_flags flags would be:
+
+ * MAVLINK_FLAG_21_CAPABLE : sender is capable of handling MAVLink 2.1
+
+Note that for the SYSID16 and TARGETTED flags the sender should only
+use the flag if is it needed. So if the systemID in the message is
+<=255 then the flag should not be set and the existing 8 bit field
```

Comment:
Just checking what happens with the ACK if a sender with system id > 255 sends a command to a target with id <255?

My understanding is the command sender would set MAVLINK_IFLAG_SYSID16 to indicate it has a 2 byte address, but not MAVLINK_IFLAG_TARGETTED because the target is in the lower range. The recipient would ignore the message if it didn't understand MAVLINK_IFLAG_SYSID16, and otherwise ack with MAVLINK_IFLAG_TARGETTED and not MAVLINK_IFLAG_SYSID16 (because it has a single byte address).

Is that right?

Reply:
yes, that's right, and it is a good case to call out in the docs as it would be easy to get wrong